### PR TITLE
[FP-12] Retail - Store Product Performance

### PR DESCRIPTION
**Pull request by arturolinares24**
@rotorrest

### CODE REVIEW

**Review by FelipeIdmeShimoku — APPROVED** [review]

File: templates/store_product_performance/app.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    and then creates and plots a new dashboard.
+    """
+    # Load environment variables
+    load_dotenv()
+
+    # Create the Shimoku client with necessary credentials
+    shimoku = Client(
+        access_token=getenv("API_TOKEN"),
+        universe_id=getenv("UNIVERSE_ID"),
+        verbosity="INFO",
```
Comment (jkahnc) [inline templates/store_product_performance/app.py:21]: In the client we don't have the "async_execution" flag activate. Don't forget to add "shimoku.run()" at the end

File: templates/store_product_performance/utils/components.py [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+        "<h1>" + title + "</h1>"
+        "<p class='base-white'>"
+        f"{subtitle}</p>"
+        "</div>"
+        "</div>"
+    )
+    return html
+
+
+def format_raw_options(sales_by_store):
```
Comment (jkahnc) [inline templates/store_product_performance/utils/components.py:61]: add return type hints

File: templates/store_product_performance/utils/utils.py [hunk 3]
```diff
@@ -0,0 +1,354 @@
+import pandas as pd
+import random
+from datetime import datetime, timedelta
+import numpy as np
+from typing import Dict, List
+import os
+
+
+def format_store_id(number):
```
Comment (jkahnc) [inline templates/store_product_performance/utils/utils.py:9]: add type hints

File: templates/store_product_performance/utils/utils.py [hunk 4]
```diff
@@ -0,0 +1,354 @@
...
+    Args:
+        number (int): Store number.
+
+    Returns:
+        str: Formatted store identifier.
+    """
+    return f"Store {number}"
+
+
+def format_product_id(number):
```
Comment (jkahnc) [inline templates/store_product_performance/utils/utils.py:22]: add type hints

File: templates/store_product_performance/utils/utils.py [hunk 5]
```diff
@@ -0,0 +1,354 @@
...
+    Args:
+        number (int): Store number.
+
+    Returns:
+        str: Formatted store identifier.
+    """
+    return f"Product {number}"
+
+
+def prepare_pivot(df, index, columns, values, unique_stores=None, unique_products=None):
```
Comment (jkahnc) [inline templates/store_product_performance/utils/utils.py:35]: add type hints

File: templates/store_product_performance/utils/utils.py [hunk 6]
```diff
@@ -0,0 +1,354 @@
...
+        df_pivot = df_pivot.reset_index()
+
+        # Format the 'Fecha' column to "Month %m of %Y"
+        df_pivot["Fecha"] = df_pivot["Fecha1"].dt.strftime("Month %m of %Y")
+        df_pivot = df_pivot.drop("Fecha1", axis=1)
+
+    return df_pivot
+
+
+def get_data(file_names: List[str]):
```
Comment (jkahnc) [inline templates/store_product_performance/utils/utils.py:91]: add return type hints

File: templates/store_product_performance/utils/utils.py [hunk 7]
```diff
@@ -0,0 +1,354 @@
...
+            "Current Year": df_yearly_pivot,
+        },
+        "Sales Product Percentage by Store": sales_product_percentage_by_store,
+        "Sales Accumulated by Product": df_all_pivot_filtered,
+    }
+
+    return results
+
+
+def get_column_name_by_value(data_dict: dict, value_to_find):
```
Comment (jkahnc) [inline templates/store_product_performance/utils/utils.py:335]: add type hints

**Review by jkahnc — APPROVED** [review]

File: templates/store_product_performance/app.py (hunk 1 above)
Comment (arturolinares24) [reply templates/store_product_performance/app.py:21]: https://github.com/shimoku-tech/shimoku-app-templates/pull/34/commits/b0233afcf2221d074646b53674fb923cdb969b83 changes made.

File: templates/store_product_performance/utils/components.py (hunk 2 above)
Comment (arturolinares24) [reply templates/store_product_performance/utils/components.py]: https://github.com/shimoku-tech/shimoku-app-templates/pull/34/commits/b0233afcf2221d074646b53674fb923cdb969b83 changes made.

File: templates/store_product_performance/utils/utils.py (hunk 3 above)
Comment (arturolinares24) [reply templates/store_product_performance/utils/utils.py]: https://github.com/shimoku-tech/shimoku-app-templates/pull/34/commits/b0233afcf2221d074646b53674fb923cdb969b83 changes made.

File: templates/store_product_performance/utils/utils.py (hunk 4 above)
Comment (arturolinares24) [reply templates/store_product_performance/utils/utils.py]: https://github.com/shimoku-tech/shimoku-app-templates/pull/34/commits/b0233afcf2221d074646b53674fb923cdb969b83 changes made.

File: templates/store_product_performance/utils/utils.py (hunk 5 above)
Comment (arturolinares24) [reply templates/store_product_performance/utils/utils.py]: https://github.com/shimoku-tech/shimoku-app-templates/pull/34/commits/b0233afcf2221d074646b53674fb923cdb969b83 changes made.

File: templates/store_product_performance/utils/utils.py (hunk 6 above)
Comment (arturolinares24) [reply templates/store_product_performance/utils/utils.py]: https://github.com/shimoku-tech/shimoku-app-templates/pull/34/commits/b0233afcf2221d074646b53674fb923cdb969b83 changes made.

File: templates/store_product_performance/utils/utils.py (hunk 7 above)
Comment (arturolinares24) [reply templates/store_product_performance/utils/utils.py]: https://github.com/shimoku-tech/shimoku-app-templates/pull/34/commits/b0233afcf2221d074646b53674fb923cdb969b83 changes made.

**Review by rotorrest — APPROVED** [review]

**Pull request merged**